src/scheduler.cc

```diff
@@ -296,12 +296,15 @@ Status SchedulerService::ExportFunction(ServerContext* context, const ExportFunc
   for (size_t i = 0; i < workers->size(); ++i) {
     ClientContext import_context;
     ImportFunctionRequest import_request;
-    import_request.mutable_function()->set_implementation(request->function().implementation());
+    import_request.mutable_function()->CopyFrom(request->function());
     if ((*workers)[i].current_task != ROOT_OPERATION) {
       ImportFunctionReply import_reply;
       (*workers)[i].worker_stub->ImportFunction(&import_context, import_request, &import_reply);
     }
   }
+  // TODO(rkn): Does this do a deep copy?
```

Comment:
It calls the copy constructor of `Function`, which is a protobuf object, and whether that copies or not depends on the default behavior of protobuf (which I assume would be to copy, but which I'm not sure of).